services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/activity.rb

```diff
@@ -35,6 +35,16 @@ def invalid_relevant_text_keys
           end.uniq
         end
 
+        def self.where_activity_scope(activity_scope)
+          ids = activity_scope
```

Comment:
You could probably keep this all in SQL, but not sure it matters:
`where(id: activity_scope.joins(:stem_vaults).select('stem_vaults.activity_id').distinct)`

Reply:
I like this idea, but the SQL ended up being more complicated than this to get it to work properly, so I'm punting on this for now. Based on the feedback, I did move it to a scope to make it's usage more clear.